### ISSUE TITLE
git-restore-mtime fails on python 3 (3.4 and 3.5)

### ISSUE THREAD

**Issue by gebailey** [MestreLion/git-tools#23]
I've packaged git-tools for Fedora and am now receiving bugzilla reports related to Python 3:
https://bugzilla.redhat.com/show_bug.cgi?id=1322472

I've created a sample git repo and am attempting to use `git-restore-mtime` on it using various python interpreters.

When running git-restore-mtime using the Python 3.4 interpreter, I get:

```
[gbailey@spock test]$ python3.4 -V
Python 3.4.4
[gbailey@spock test]$ python3.4 /usr/bin/git-restore-mtime 
Traceback (most recent call last):
  File "/usr/bin/git-restore-mtime", line 165, in <module>
    lsfileslist.add(os.path.relpath(line.strip(), workdir))
  File "/usr/lib64/python3.4/posixpath.py", line 448, in relpath
    start_list = [x for x in abspath(start).split(sep) if x]
TypeError: Can't convert 'bytes' object to str implicitly
```

When running git-restore-mtime using the Python 3.5 interpreter, I get:

```
[gbailey@spock test]$ python3.5 -V
Python 3.5.1
[gbailey@spock test]$ python3.5 /usr/bin/git-restore-mtime 
Traceback (most recent call last):
  File "/usr/bin/git-restore-mtime", line 165, in <module>
    lsfileslist.add(os.path.relpath(line.strip(), workdir))
  File "/usr/lib64/python3.5/posixpath.py", line 459, in relpath
    genericpath._check_arg_types('relpath', path, start)
  File "/usr/lib64/python3.5/genericpath.py", line 145, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None
TypeError: Can't mix strings and bytes in path components
```

**Comment by Coacher** [comment]
Same here on Gentoo linux.

**Comment by MestreLion** [comment]
Fixed by PR #26
